services/src/main/java/io/scalecube/services/gateway/Gateway.java

```diff
@@ -31,13 +31,20 @@ static Gateway getGateway(Class<? extends Gateway> gatewayClass) {
    * @param call service call definition
    * @param metrics @return IP socket address on which gateway is listening to requests
    */
-  Mono<InetSocketAddress> start(
+  Mono<Gateway> start(
       GatewayConfig config,
       Executor workerThreadPool,
       boolean preferNative,
       Call call,
       Metrics metrics);
 
+  /**
+   * Returns Gateway's address if it's started, {@code null} otherwise.
+   *
+   * @return Mono of listening address of gateway if it's started, otherwise - empty.
```

Comment:
> otherwise - empty

Nope. Never null. We just shouldn't give GW to start if it ccan't bind.

Reply:
done